### PR TITLE
Fixes violations of Lint/UnusedBlockArgument and reenables the rubocop check.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Initial fixes generated by "bundle exec rubocop --only Lint/UnusedBlockArgument --auto-correct".

Further simplification done manually.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Is this a variable naming convention that we want to enforce?

File: aws/build.rake [hunk 1]
```diff
@@ -68,7 +68,7 @@ end
 #
 def create_threads(count)
   [].tap do |threads|
-    (1..count).each do |i|
+    (1..count).each do |_i|
```
Comment (joshlory) [inline aws/build.rake]: Nit: clearer as `count.times do`?

**Comment by joshlory** [comment]
In places where there is no used parameter following the unused parameters, does it make sense to remove the unused parameters?

As for naming, I've seen it written as `{ |_a, _b, c| ... }` and as `{ |_, _, c| ... }`.  I think the first makes sense if the names of the skipped parameters are important.  The second is what RubyMine recommends and is more concise.  No strong preference here.

File: aws/build.rake (hunk 1 above)
Comment (ashercodeorg) [inline aws/build.rake]: Absolutely (in another PR).

**Comment by ashercodeorg** [comment]
Removing unused parameters: By and large, I don't have a strong opinion. Is there a rubocop that enforces their removal? If so, I'll get to it when I get to that check.

As for naming, I'll stick with the auto-generated stuff if you don't have a strong preference.

**Comment by joshlory** [comment]
This is the cop that would enforce their removal.  If you have an unused param, you can remove it, or prefix it with `_` as a convention to say "yes I know this param isn't used".  Unused params at the end of the list can (usually) just be dropped instead of prefixing them:

``` ruby
> (1..3).each { puts 'hello' } # Range#each doesn't care if you omit the block argument
hello
hello
hello
> (1..3).each { |_i| puts 'hello' }
hello
hello
hello
```

But this isn't _always_ the case.  Methods can see how many arguments are passed to the block via `block.arity`.  For example `Hash#each` has different behavior with one block argument vs. two:

``` ruby
> h = {a: 1, b: 2}
> h.each{ |k| puts k }
a
1
b
2
> h.each{ |k, _| puts k }
a
b
```

Better to remove ambiguity with:

``` ruby
> h.each_key{ |k| puts k }
a
b
```

Since we're re-enabling the rule, now is the best time to audit which instances should be kept and prefixed, and which instances can be dropped.  Otherwise the `--auto-correct` will cover up the warnings by doing the "safe" thing and prefixing all the unused params.

File: lib/dynamic_config/adapters/memory_adapter.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@ def get(key)
 
   def all
     ret = {}
-    @hash.each do |k, v|
+    @hash.each do |k, _v|
```
Comment (ashercodeorg) [inline lib/dynamic_config/adapters/memory_adapter.rb]: Changing manually to @hash.each_key do |k|.

File: i18n/code.org/lib/copy-untranslated-apps.rb [hunk 3]
```diff
@@ -76,7 +76,7 @@
   'netsim'
 ]
 
-locales.each_pair do |language, locale|
+locales.each_pair do |_language, locale|
```
Comment (ashercodeorg) [inline i18n/code.org/lib/copy-untranslated-apps.rb]: Changing manually to locales.each_value do |locale|.

Comment (ashercodeorg) [inline i18n/code.org/lib/copy-untranslated-apps.rb]: Changing manually to locales.each_value do |locale|.

File: i18n/hourofcode.com/out.rb [hunk 4]
```diff
@@ -118,7 +118,7 @@
 #########################################################################################
 
 puts "Copying files from cdo/i18n to hoc.com/i18n..."
-languages.each_pair do |name, codes|
+languages.each_pair do |_name, codes|
```
Comment (ashercodeorg) [inline i18n/hourofcode.com/out.rb]: Changing manually to languages.each_value do |codes|.

File: aws/build.rake (hunk 1 above)
Comment (ashercodeorg) [inline aws/build.rake]: Done (in this PR).

File: bin/deliver_poste_messages [hunk 5]
```diff
@@ -251,7 +251,7 @@ end
 
 def create_threads(count)
   [].tap do |threads|
-    (1..count).each do |i|
+    (1..count).each do |_i|
```
Comment (ashercodeorg) [inline bin/deliver_poste_messages]: Manually changed to 'count.times do'.

**Comment by ashercodeorg** [comment]
Good thoughts. :) Made a bunch of manual changes, as noted. Please take another look.

File: dashboard/test/ui/support/hooks.rb [hunk 6]
```diff
@@ -1,6 +1,6 @@
 saw_failure = false
 
-Before('@stop_after_failure') do |scenario|
+Before('@stop_after_failure') do |_scenario|
```
Comment (joshlory) [inline dashboard/test/ui/support/hooks.rb]: Nit: looks like this argument is truly optional (see line 15).

File: dashboard/app/models/concerns/serialized_properties.rb [hunk 7]
```diff
@@ -51,7 +51,7 @@ def define_methods_for_property(property_name)
     def define_methods_for_encrypted_property(property_name)
       cleartext_property_name = property_name.gsub(ENCRYPTED_PROPERTY_REGEX, '')
 
-      define_method(cleartext_property_name) do |*args|
+      define_method(cleartext_property_name) do |*_args|
```
Comment (joshlory) [inline dashboard/app/models/concerns/serialized_properties.rb]: Nit: block argument seems to be optional (see http://ruby-doc.org/core-2.2.0/Module.html#method-i-define_method).

File: dashboard/config/initializers/mini_profiler.rb [hunk 8]
```diff
@@ -13,7 +13,7 @@
   # RACK_MINI_PROFILER environment variable is 'on'. This can be done either by
   # setting the environment variable prior to running the server, or by hitting
   # a page with a "pp=enabled" query string parameter and then refreshing the page.
-  Rack::MiniProfiler.config.pre_authorize_cb = lambda {|env|
+  Rack::MiniProfiler.config.pre_authorize_cb = lambda {|_env|
     ENV['RACK_MINI_PROFILER'] == 'on'
```
Comment (joshlory) [inline dashboard/config/initializers/mini_profiler.rb:17]: I wonder if this is an actual "bug", i.e. should we be setting `env['RACK_MINI_PROFILER']` instead?

File: dashboard/test/ui/support/hooks.rb (hunk 6 above)
Comment (ashercodeorg) [inline dashboard/test/ui/support/hooks.rb]: Done (removed).

File: dashboard/app/models/concerns/serialized_properties.rb (hunk 7 above)
Comment (ashercodeorg) [inline dashboard/app/models/concerns/serialized_properties.rb]: Done.

**Comment by joshlory** [comment]
Feel free to ignore the nits.  LGTM!

File: dashboard/config/initializers/mini_profiler.rb (hunk 8 above)
Comment (ashercodeorg) [inline dashboard/config/initializers/mini_profiler.rb:17]: Good catch. Let me ask Will (who might have the most context).

**Pull request merged**